src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -646,14 +637,45 @@ private bool InitProcessPing(string targetNameOrAddress, out string resolvedTarg
         {
             resolvedTargetName = targetNameOrAddress;
 
-            IPHostEntry hostEntry;
+            IPHostEntry hostEntry = null;
             if (IPAddress.TryParse(targetNameOrAddress, out targetAddress))
             {
-                if (ResolveDestination)
+                if ((IPv4 && targetAddress.AddressFamily != AddressFamily.InterNetwork)
+                    || (IPv6 && targetAddress.AddressFamily != AddressFamily.InterNetworkV6))
```

Comment:
It is wrong scenario with wrong result (See below - 10.0.75.1 is not my address).
I suggest remove this because it confuse a lot.

```
Test-Connection -IPv4 fe80::5034:5357:6b2a:9c3d


   Destination: 10.0.75.1

Ping Source           Address                   Latency BufferSize Status
                                                   (ms)        (B)
---- ------           -------                   ------- ---------- ------
   1 Sazonov1         10.0.75.1                       0         32 Success
   2 Sazonov1         10.0.75.1                       0         32 Success
   3 Sazonov1         10.0.75.1                       0         32 Success
   4 Sazonov1         10.0.75.1                       0         32 Success

Reply:
Is there an alternative? Should we simply throw an error in the case the user provides an address of the wrong family?

My understanding is that this path should do reverse lookup for the address given and retrieves the first ipaddress of the correct family from the DNS information.

This may not reflect exactly the same address and is dependent on the DNS response. I am inclined to think it may be of use in edge cases and the limitation should merely be documented.

But if you think we should error out instead, I can make that change.

Reply:
I think we could look on ping behavior:
```
C:\WINDOWS\system32>ping -4 fe80::7d44:95a0:9876:9fbf
Ping request could not find host fe80::7d44:95a0:9876:9fbf. Please check the name and try again.
```

Reply:
That seems sufficiently logical to me. I did rather like this feature, but I can understand that it's not the most reliable thing to work with.

Reply:
If the scenario important (I don't think so) it should be implemented under specific parameter. It is out of the PR.